documentation/writing_tests.md

```diff
@@ -0,0 +1,1072 @@
+
+
+
+## Recommended tools
+
+Writing tests for JavaScript and TypeScript requires testing tools, such as a test framework, an assertion library, and a way to bundle and run these tests in various environments. The JavaScript community has many overlapping tools that one could pick to fulfill any of these tasks. To ensure that the testing experience across the Azure SDK for JS/TS is consistent and reliable, we've picked the following external testing tools:
+
+- [Mocha](https://www.npmjs.com/package/mocha), which offers a well known and stable test framework for both NodeJS and the browser.
+- [Chai](https://www.npmjs.com/package/chai), a well known assertion library for Node and the browser.
+- [Rollup](https://www.npmjs.com/package/rollup), to bundle JavaScript for different environments, which helps us write a single TypeScript source and trust it will compile correctly for Node and the browser.
+- [Karma](https://www.npmjs.com/package/karma), which allows us to run our tests in multiple browsers.
+
+We've also come up with our own internal tools:
+
+- The [Recorder](https://github.com/Azure/azure-sdk-for-js/tree/master/sdk/test-utils/recorder) is a tool that helps us run our live tests against static recordings obtained from a previous successful run, which aims to ensure that our code hasn't changed while benefitting from not having to reach out to live services.
+
+Now, let's see how we're using the mentioned tools.
+
+### Mocha
+
+Mocha is a well known JavaScript test framework that runs in NodeJS and in the browser. If you're new to Mocha, we recommend you to dive into [Mocha's documentation](https://mochajs.org/).
+
+In this section, we'll explore how the Azure SDK for JS is using mocha, what versions we're using, how we're configuring it, and other recommendations on how to use this library.
+
+#### Mocha in our dependencies
+
+Since we're using [Rush](https://rushjs.io/), we're forced to use the same version of our packages in each one of the projects inside of this repository. If you want to add `mocha` as a dev dependency to a new project inside of this repository, first make sure you are in the root folder of that project, then you can run the following command:
+
+```
+rush add --dev -p mocha
+```
+
+We're using the following Mocha plugins and related dependencies:
+
+- [`@types/mocha`](https://www.npmjs.com/package/@types/mocha) provides the type definitions for mocha.
+- [`mocha-multi`](https://www.npmjs.com/package/mocha-multi) to let Mocha use multiple reporters.
```

Comment:
we no longer use `mocha-multi` in our tests thanks to @HarshaNalluru making a custom reporter for us

Reply:
Thanks! I removed this